### PR TITLE
New Formula: Widelands

### PR DESCRIPTION
**Pull request by s172262**
I'm sorry to break previous pull request #543.

### CODE REVIEW

File: widelands.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  depends_on "sdl_mixer"
+  depends_on "sdl_net"
+  depends_on "sdl_ttf"
+  depends_on "sdl_gfx"
+  depends_on "doxygen"
+  depends_on "glew"
+  depends_on "lua51"
+
+  def install
+    ENV.append "CPPFLAS", "-I#{Formula["gettext"].opt_prefix}/include -I#{Formula["lua51"].opt_prefix}/include/lua-5.1"
```
Comment (dunn) [inline widelands.rb]: Should've caught this last time, but you can use `opt_include` and `opt_lib` here and below to save space.

Comment (dunn) [inline widelands.rb]: But are you sure these two lines are necessary? It builds for me without them.

File: widelands.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    ENV.append "LDFLAGS", "-L#{Formula["gettext"].opt_prefix}/lib -L#{Formula["lua51"].opt_prefix}/lib"
+
+    cd "build" do
+      system "cmake", "..", "-DLUA_INCLUDE_DIR:PATH=#{Formula["lua51"].opt_prefix}/include/lua-5.1",
+                      "-DLUA_LIBRARIES:STRING=#{Formula["lua51"].opt_prefix}/lib/liblua.5.1.dylib",
+                      "-DLUA_LIBRARY:FILEPATH=#{Formula["lua51"].opt_prefix}/lib/liblua.5.1.dylib",
+                      "-DINTL_INCLUDE_DIR:PATH=#{Formula["gettext"].opt_prefix}/include",
```
Comment (dunn) [inline widelands.rb]: It seems to build and run fine without this line or the `libpng` one below as well.

File: widelands.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                      # Without the following option, Cmake uses the library of framework.
+                      "-DPNG_PNG_INCLUDE_DIR:PATH=#{Formula["libpng"].opt_prefix}/include",
```
Comment (s172262) [inline widelands.rb]: In my local Mac libpng**14** of Mono framework was linked without this line.

File: widelands.rb (hunk 2 above)
Comment (s172262) [inline widelands.rb]: I think that it is necessary because gettext is `keg_only`.

File: widelands.rb (hunk 1 above)
Comment (s172262) [inline widelands.rb]: I think that it is necessary because gettext is `keg_only`.

File: widelands.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  depends_on "sdl_net"
+  depends_on "sdl_ttf"
+  depends_on "sdl_gfx"
+  depends_on "doxygen"
+  depends_on "glew"
+  depends_on "lua51"
+
+  def install
+    ENV.append "CPPFLAS", "-I#{Formula["gettext"].opt_prefix}/include -I#{Formula["lua51"].opt_prefix}/include/lua-5.1"
+    ENV.append "LDFLAGS", "-L#{Formula["gettext"].opt_prefix}/lib -L#{Formula["lua51"].opt_prefix}/lib"
```
Comment (s172262) [inline widelands.rb]: In my local Mac Cmake linkded with #{HOMEBREW_PREFIX}/lib/liblua.dylib which is symbolic link of liblua.5.3.dylib without  "-L#{Formula["lua51"].opt_prefix}/lib"
Formula of lua51 didn't have liblua.dylib.

File: widelands.rb (hunk 1 above)
Comment (s172262) [inline widelands.rb]: Cmake couldn't find out lua.hpp . So I think that Cmake didn't see #{Formula["lua51"].opt_include}/lua5.1 or #{Formula["lua51"].opt_include}/lua-5.1.

File: widelands.rb (hunk 2 above)
Comment (dunn) [inline widelands.rb]: Homebrew adds all the dependencies to the required `PATH`s, so CMake should be able to find it.

File: widelands.rb (hunk 4 above)
Comment (dunn) [inline widelands.rb]: Oh got it, with `lua53` installed it's picking up that instead.  But we should be able to remove the `gettext` stuff.

**Comment by apjanke** [comment]
@BrewTestBot test this please

**Comment by apjanke** [comment]
Merged. Thank you for your contribution to Homebrew!

For future reference, you can squash your commits to a single commit (to keep history tidy) with `git rebase -i master` or `git commit --amend`.

**Pull request closed without merging**